### PR TITLE
Tim/pod update+objc protocol fix

### PR DESCRIPTION
**Pull request by TG908**
I've fixed a number of warnings, updated function signatures (objc compatibility), updated some pods, updated the xcode project settings, enabled wmo and some pods are now compiled for swift 4 while others are compiled for swift 3.3

This should work with Swift 4.0.3 (Xcode 9.2) [download](https://swift.org/builds/swift-4.0.3-release/xcode/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-osx.pkg).

### CODE REVIEW

**Comment by TG908** [comment]
ERROR: Error during SonarQube Scanner execution
ERROR: Failed to execute project builder: org.sonar.plugins.github.PullRequestProjectBuilder
ERROR: Caused by: Unable to perform GitHub WS operation: {"message":"Bad credentials","documentation_url":"https://developer.github.com/v3"}

File: Podfile [hunk 1]
```diff
@@ -26,9 +26,28 @@ target 'Campus' do
...
 post_install do |installer|
+    
+    targets_swift_4 = ['Kanna','TKSubmitTransition','SWXMLHash','CalendarLib','ASWeekSelectorView','PZPullToRefresh','AYSlidingPickerView','Fuzzi']
+    
     installer.pods_project.targets.each do |target|
-        target.build_configurations.each do |config|
-            config.build_settings['SWIFT_VERSION'] = '3.2'
+        if targets_swift_4.include? target.name
+            target.build_configurations.each do |config|
+                config.build_settings['SWIFT_VERSION'] = '4.0'
```
Comment (mathiasquintero) [inline Podfile]: these only defer on this line. Move the if here to remove duplicated code

File: TUM Campus App/CafeteriaViewController.swift [hunk 2]
```diff
@@ -151,15 +151,15 @@ extension CafeteriaViewController {
 
 extension CafeteriaViewController: ASWeekSelectorViewDelegate {
     
-    func weekSelector(_ weekSelector: ASWeekSelectorView!, didSelect date: Date!) {
+    func weekSelector(_ weekSelector: ASWeekSelectorView, didSelect date: Date) {
```
Comment (mathiasquintero) [inline TUM Campus App/CafeteriaViewController.swift]: Have you tried these out? To check if they're getting called

**Comment by mathiasquintero** [comment]
I will later check how I can get this on the 4.1 toolchain. I just have to update Xcode first ;)

File: TUM Campus App/CafeteriaViewController.swift (hunk 2 above)
Comment (TG908) [reply TUM Campus App/CafeteriaViewController.swift]: They are all getting called. I checked with breakpoints.

File: Podfile (hunk 1 above)
Comment (TG908) [reply Podfile]: done

**Comment by TG908** [comment]
included in #251

**Comment by thellmund** [comment]
Builds flawlessly 👍

**Pull request merged**